### PR TITLE
Bugfix/customer io race condition

### PR DESCRIPTION
**Pull request by sbilling**
QA instructions:

1. make sure you are signed out
2. via any newsletterCallout i.e. an email input with a "subscribe" button, enter an email for a user you know does not exist
3. observe, toast confirms that subscription prefs were saved
4. repeat 2 with a user you know exists
5. observe, toast confirms that subscription prefs were saved
6. confirm customer records exist in customer.io

### CODE REVIEW

File: lib/utils/supabase-admin.ts [hunk 1]
```diff
@@ -156,11 +156,22 @@ const createAuthUser = async (email: string, full_name?: string) => {
...
   if (error) {
-    return { data: null, error };
+    // If we hit this error, its because the user already exists.
+    return { data: null, shouldAwaitNewUserLatency: false, error };
   }
-  console.log(`Auth user created: ${user?.id}`);
-  return { data: user, error: null };
+
+  // Create the customer.io record to avoid race condition
+  await fetch(`${process.env.SITE_URL}/api/customer-io/create-customer`, {
```
Comment (mattThousand) [inline lib/utils/supabase-admin.ts:166]: I think we're making an unnecessary network call to our own server. Could you create  a `CustomerIoService`, export `createCioCustomer` from it and call it directly?

Comment (mattThousand) [reply lib/utils/supabase-admin.ts:166]: @sbilling 
[flatbread.patch](https://github.com/Pirate-Wires/pirate-wires/files/14172135/flatbread.patch)

Comment (mattThousand) [reply lib/utils/supabase-admin.ts:166]: @sbilling 
[flatbread.patch](https://github.com/Pirate-Wires/pirate-wires/files/14172280/flatbread.patch)

Comment (mattThousand) [reply lib/utils/supabase-admin.ts:166]: [flatbread.patch](https://github.com/Pirate-Wires/pirate-wires/files/14172333/flatbread.patch) 🫓

File: components/newsletterCallout.tsx [hunk 2]
```diff
@@ -72,10 +72,12 @@ export default function NewsletterCallout({ newsletterData, globalFields }) {
     if (!email) {
       setIsSuccess(false);
       setError(new ToastableError("Email is required"));
+      return;
     }
     if (!selectedNewsLetters.length) {
       setIsSuccess(false);
       setError(new ToastableError("Please select at least one newsletter"));
+      return;
```
Comment (mattThousand) [inline components/newsletterCallout.tsx:80]: Maybe throw an error toast instead? User might not know why they're getting bonked

Comment (sbilling) [reply components/newsletterCallout.tsx:80]: It's on the line above it

Comment (mattThousand) [reply components/newsletterCallout.tsx:80]: I'll see myself out 😅

**Review by mattThousand — APPROVED** [review]

**Pull request merged**